### ISSUE TITLE
Remove the DatePicker component code, and replace its usage with the community DateTimePicker

### ISSUE THREAD

**Issue by NickGerleman** [microsoft/react-native-windows#4221]
Upstream React native provides their DatePicker component as DatePickerAndroid and DatePickerIOS. We deviate by calling ours DatePicker. We should rename our component.

Note that this component was moved out of RN, and into https://github.com/react-native-community/react-native-datetimepicker#react-native-datetimepicker

**Comment by chrisglein** [comment]
Question here is whether we migrate our DatePicker out to the community module in order to fix this. DatePickerAndroid and DatePickerIOS are moving out of lean core, so we need to do that anyway. If we do all that, we don't need the rename.

**Comment by NickGerleman** [comment]
Makes sense. The only thing hairy here is that it's not fully removed from core yet, only deprecated. Likely not worth making a breaking change right before removing though if we are able to move it.

**Comment by chrisglein** [comment]
Was removed in 0.60: https://reactnative.dev/docs/0.60/datepickerios
(not present in 0.61)
So really, we should already have this removed...

**Comment by NickGerleman** [comment]
The doc linked mentioned they were planning on removing it. The source is still there though on 61. https://github.com/facebook/react-native/tree/0.61-stable/Libraries/Components/DatePicker

**Comment by NickGerleman** [comment]
Current docs still say it will be removed in a future release https://reactnative.dev/docs/datepickerios

The files are still there for master, so the earliest it will be removed is 63.

**Comment by chrisglein** [comment]
Related #4156

**Comment by rectified95** [comment]
I'm working on porting the DatePicker to the community repo. I've noticed that the current Windows implementation only provides the date picker, while this component also exposes the time selector on Android/iOS as the second mode of operation. Should I investigate adding that functionality as part of this work, or create a separate issue to track it? This would mean that the Windows version would be temporarily poorer than its counterparts in the community repo, but it wouldn't strictly be a regression. @chrisglein @NickGerleman 

FYI @robertev

**Comment by chrisglein** [comment]
> Should I investigate adding that functionality as part of this work, or create a separate issue to track it? 

Create a separate issue to track it. Matching the behavior we have in core is the main priority because it unblocks us from removing it from core. Hopefully the repo owners are okay taking a partial implementation. If they aren't we can stage it in a branch and work on filling out time picker stuff.

**Comment by rectified95** [comment]
I've added Windows support to the community module.
I will use this issue to track the work of removing the existing implementation from this repo.
Also, renaming this issue, since the compoinent's name has for some time been RNDateTimePicker/DateTimePicker, not DatePickerAndroid and DatePickerIOS.
Fyi @NickGerleman

**Comment by rectified95** [comment]
Forgot to comment before earlier: DatePickerAndroid/iOS are still in the RN repo, even though they're deprecated; listed as "to do" - https://github.com/facebook/react-native/issues/23313
so I cant remove it from this repo until that changes.

**Comment by NickGerleman** [comment]
We should be able to remove this one at any time since our API was different from the one in core anyway.

**Comment by acoates-ms** [comment]
We should have a plan to start producing warnings one release, then soft errors, then delete it?

**Comment by NickGerleman** [comment]
Uggh, you're right. DatePickerIOS, etc are producing warnings on require but our DatePicker isn't yet.

**Comment by NickGerleman** [comment]
@rectified95 would you be able to add a warning in `vnext\index.windows.js` when `DatePicker` is required, similar to what is present for `DatePickerIOS`, etc?

**Comment by rectified95** [comment]
Sure

**Comment by rectified95** [comment]
In order to move things forward, I have created a PR removing DatePicker code from RN core - https://github.com/facebook/react-native/pull/30576
Unfortunately, it got closed, along with the entire master issue tracking Lean Core progress:
https://github.com/facebook/react-native/issues/23313#issuecomment-746121956

Apparently, Facebook uses the DatePicker internally - probably alongside Picker and Slider, which I also started removing - so they want to do the removal themselves at some point.

Once that happens, we will ingest the change in one of our integrations - closing this issue and getting back to the pending PR adding deprecation warnings.

**Comment by robertev** [comment]
Interesting, I'll pass that onto my customer.

**Comment by rectified95** [comment]
@robertev FYI - I added a Windows implementation to the community module a while ago - that's what should be consumed. Your customer shouldn't be affected by this issue, or use the deprecated version.

https://github.com/react-native-datetimepicker/datetimepicker

**Comment by tamarNinaLevy** [comment]
hi

**Comment by tamarNinaLevy** [comment]
i am working on a react native app and about a week ago i started geting these warnings and i dont know how to solve it:
Error: Unable to resolve module ./Libraries/Components/DatePicker/DatePickerIOS from /home/hilma/projects/tell-me/tellme-client/node_modules/react-native/index.js: 

None of these files exist:
  * node_modules/react-native/Libraries/Components/DatePicker/DatePickerIOS(.native|.native.ts|.ts|.native.tsx|.tsx|.native.js|.js|.native.jsx|.jsx|.native.json|.json|.native.svg|.svg)
  * node_modules/react-native/Libraries/Components/DatePicker/DatePickerIOS/index(.native|.native.ts|.ts|.native.tsx|.tsx|.native.js|.js|.native.jsx|.jsx|.native.json|.json|.native.svg|.svg)
  15 | import typeof ActivityIndicator from './Libraries/Components/ActivityIndicator/ActivityIndicator';
  16 | import typeof Button from './Libraries/Components/Button';
> 17 | import typeof DatePickerIOS from './Libraries/Components/DatePicker/DatePickerIOS';
     |                                   ^
  18 | import typeof DrawerLayoutAndroid from './Libraries/Components/DrawerAndroid/DrawerLayoutAndroid';
  19 | import typeof FlatList from './Libraries/Lists/FlatList';
  20 | import typeof Image from './Libraries/Image/Image';
    at ModuleResolver.resolveDependency (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:114:15)
    at DependencyGraph.resolveDependency (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/node-haste/DependencyGraph.js:281:43)
    at Object.resolve (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/lib/transformHelpers.js:170:21)
    at Graph._resolveDependencies (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/DeltaBundler/Graph.js:473:35)
    at Graph._processModule (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/DeltaBundler/Graph.js:261:38)
    at async Graph._addDependency (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/DeltaBundler/Graph.js:372:20)
    at async Promise.all (index 1)
    at async Graph._processModule (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/DeltaBundler/Graph.js:322:5)
    at async Graph._addDependency (/home/hilma/projects/tell-me/tellme-client/node_modules/metro/src/DeltaBundler/Graph.js:372:20)
    at async Promise.all (index 1)
Error: Unable to resolve module ./Libraries/Components/DatePicker/DatePickerIOS from /home/hilma/projects/tell-me/tellme-client/node_modules/react-native/index.js: 

it has got something to do with the library of datePickerios but i dont recall installing it however i am using activity indicator from react native in a few components, i looked into the activity indicators type and from wha i understood it had nothing to dowith date
pickerios, i saw in the documentation that it has been deprecated but why do i get the warning if iam not using it anywhere.

id be grateful to get an answer.

**Comment by tamarNinaLevy** [comment]
this is my package.json:
{
  "name": "tellme-client",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "proxy": "https://tell-me-t.carmel6000.com",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@expo-google-fonts/rubik": "^0.2.2",
    "@hilma/auth": "^1.0.1",
    "@hilma/auth-native": "^0.7.1",
    "@hilma/fileshandler-native": "^0.1.19",
    "@hilma/tools": "^0.2.0",
    "@pietile-native-kit/keyboard-aware-scrollview": "^1.5.0",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@react-native-clipboard/clipboard": "^1.11.2",
    "@react-native-community/audio-toolkit": "^2.0.3",
    "@react-native-community/blur": "^4.3.0",
    "@react-native-community/clipboard": "^1.5.1",
    "@react-native-google-signin/google-signin": "^9.0.2",
    "@react-navigation/drawer": "^6.5.6",
    "@react-navigation/native": "^6.1.1",
    "@react-navigation/stack": "^6.3.10",
    "axios": "^1.3.3",
    "expo": "^47.0.13",
    "expo-app-loading": "~2.1.1",
    "expo-apple-authentication": "^5.0.1",
    "expo-auth-session": "^3.8.0",
    "expo-av": "~13.0.3",
    "expo-blur": "~12.0.1",
    "expo-clipboard": "~4.0.1",
    "expo-constants": "~14.0.2",
    "expo-device": "~5.0.0",
    "expo-document-picker": "~11.0.1",
    "expo-linear-gradient": "~12.0.1",
    "expo-random": "~13.0.0",
    "expo-screen-orientation": "~5.0.1",
    "expo-speech": "~11.0.0",
    "expo-status-bar": "~1.4.2",
    "expo-svg-uri": "^1.3.1",
    "expo-updates": "~0.15.6",
    "expo-web-browser": "~12.0.0",
    "fs-promise": "^2.0.3",
    "jwt-decode": "^3.1.2",
    "link": "^1.5.1",
    "lodash.isequal": "^4.5.0",
    "native-base": "^3.4.23",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-native": "0.70.8",
    "react-native-config": "^1.4.11",
    "react-native-elements": "^3.4.2",
    "react-native-extended-stylesheet": "^0.12.0",
    "react-native-gesture-handler": "~2.8.0",
    "react-native-input-scroll-view": "^1.11.0",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-paper": "^4.12.5",
    "react-native-reanimated": "~2.12.0",
    "react-native-safe-area-context": "4.4.1",
    "react-native-screens": "~3.18.0",
    "react-native-select-dropdown": "^3.1.0",
    "react-native-sound": "^0.11.2",
    "react-native-svg": "13.4.0",
    "react-native-svg-transformer": "^1.0.0",
    "react-native-vector-icons": "^9.2.0",
    "react-native-vector-image": "^0.3.3",
    "react-native-viewport-units": "^0.0.5",
    "react-native-web": "~0.18.9",
    "react-native-webview": "11.23.1",
    "react-native-wheel-color-picker": "^1.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@types/lodash": "^4.14.191",
    "@types/lodash.isequal": "^4.5.6",
    "@types/react": "~18.0.14",
    "@types/react-native": "^0.71.6",
    "@types/react-native-vector-icons": "^6.4.12",
    "react-native-svg-transformer": "^1.0.0",
    "typescript": "^4.6.3"
  },
  "private": true
}
